src/mcrypt/ctr.c

```diff
@@ -93,14 +93,20 @@ GF_Err _mcrypt_set_state(void *_buf, void *IV, int size)
 
 GF_Err _mcrypt_get_state(void *buf, void *IV, int *size)
 {
-	if (*size < ((CTR_BUFFER* )buf)->blocksize + 1) {
-		*size = ((CTR_BUFFER* )buf)->blocksize + 1;
+	if (*size < ((CTR_BUFFER* )buf)->blocksize) { 
+		*size = ((CTR_BUFFER* )buf)->blocksize;
 		return GF_BAD_PARAM;
 	}
-	*size = ((CTR_BUFFER* )buf)->blocksize + 1;
```

Comment:
Perhaps the variable name in getter/setter routines make a confusion : the IV in set state and get state is in fact the current counter position + IV. In setter is copy first byte to c_counter_pos and to c_counter (i.e the counter which is used in encryption/decryption) so any byte was skipped.